test/createAzureResource.test.ts

```diff
@@ -6,30 +6,32 @@
 import * as assert from 'assert';
 import { ResourceManagementClient } from 'azure-arm-resource';
 import { WebSiteManagementClient, WebSiteManagementModels } from 'azure-arm-website';
+import { execSync } from 'child_process';
+import * as fse from 'fs-extra';
 import { IHookCallbackContext, ISuiteCallbackContext } from 'mocha';
 import * as vscode from 'vscode';
-import { AzExtTreeDataProvider, AzureAccountTreeItemWithProjects, DialogResponses, ext, getGlobalSetting, getRandomHexString, ProjectLanguage, projectLanguageSetting, TestAzureAccount, TestUserInput, updateGlobalSetting } from '../extension.bundle';
-import { longRunningTestsEnabled } from './global.test';
+import { AzExtTreeDataProvider, AzureAccountTreeItemWithProjects, delay, DialogResponses, ext, FunctionTreeItem, getRandomHexString, IActionContext, ProjectLanguage, ProjectRuntime, TestAzureAccount, TestUserInput } from '../extension.bundle';
+import { getTestRootFolder, longRunningTestsEnabled } from './global.test';
 import { runWithFuncSetting } from './runWithSetting';
+import { getCSharpValidateOptions, getJavaScriptValidateOptions, validateProject } from './validateProject';
 
 // tslint:disable-next-line:max-func-body-length
 suite('Create Azure Resources', async function (this: ISuiteCallbackContext): Promise<void> {
     this.timeout(1200 * 1000);
     const resourceGroupsToDelete: string[] = [];
     const testAccount: TestAzureAccount = new TestAzureAccount();
-    let oldProjectLanguage: ProjectLanguage | undefined;
     let webSiteClient: WebSiteManagementClient;
     const resourceName1: string = getRandomHexString().toLowerCase();
+    // Get the *.code-workspace workspace file path
+    const projectPath: string = getTestRootFolder();
+    // tslint:disable-next-line: prefer-const
+    let context: IActionContext | undefined;
```

Comment:
I want to wait until Eric comes back from being OOF, but I think that this is not a good solution to not having the `IActionContext`.  I'll discuss with him offline and then update this comment.

Reply:
I think the IActionContext does not actually sense, so handled it like this. But you should be right, so i wait for your reply. Thanks a lot.